test/test.js

```diff
@@ -613,6 +698,36 @@ describe('Filter', function() {
     write(fileForChange, 'Nicest cats in need of homes');
   }));
 
+  it('replaces stale entries - async', co.wrap(function* () {
+    let fileForChange = path.join(fixturePath('a'), 'dir', 'a', 'README.md');
+
+    let subject = new ReplaceAsyncFilter(fixturePath('a'), {
+      glob: '**/*.md',
+      search: 'dogs',
+      replace: 'cats',
+      async: true,
+    });
+    let output = createBuilder(subject);
+
+    yield output.build();
+
+    expect(file(fileForChange)).to.exist;
+
+    write(fileForChange, 'such changes');
+
+    expect(file(fileForChange)).to.exist;
+
+    yield output.build();
+
+    expect(file(fileForChange)).to.exist;
+
+    write(fileForChange, 'such changes');
+
+    expect(file(fileForChange)).to.exist;
+
+    write(fileForChange, 'Nicest cats in need of homes');
```

Comment:
I copy-pasted this from the other test, without enough of a critical eye. It's changing the fixture file back at the end of the test, which won't happen if an expectation fails, and will affect other tests. I changed this to update that file in `afterEach()` and provide better isolation for these tests.